app/Jobs/SetWikiLogo.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Wiki;
+use App\WikiSetting;
+use Illuminate\Contracts\Filesystem\Cloud;
+use Illuminate\Database\QueryException;
+use Illuminate\Http\File;
+use Intervention\Image\Facades\Image;
+use Storage;
+
+/**
+ * This can be run with the artisan job command, for example:
+ * php artisan job:dispatchNow SetWikiLogo domain wiki.addshore.com /path/to/logo.png
+ * php artisan job:dispatchNow SetWikiLogo id 1234 /path/to/logo.png
+ * 
+ * NOTE: This job needs to be run as the correct user if run via artisan (instead of via the UI)
+ */
+class SetWikiLogo extends Job
+{
+    private $wikiKey;
+    private $wikiValue;
+    private $logoPath;
+
+    /**
+     * @return void
+     */
+    public function __construct(string $wikiKey, string $wikiValue, string $logoPath)
```

Comment:
Don't need type hint for constructor

```suggestion
    public function __construct(string $wikiKey, string $wikiValue, string $logoPath)
```